fv3core/grid/generation.py

```diff
@@ -0,0 +1,2136 @@
+        """
+        Sine of the angle at point 2 of the 'supergrid' that refines each grid cell:
+        9---4---8
+        |       |
+        1   5   3
+        |       |
+        6---2---7
+        """
+        if self._sin_sg2 is None:
+            self._init_cell_trigonometry()
+        return self._sin_sg2
+
+    @property
+    def sin_sg3(self):
+        """
+        Sine of the angle at point 3 of the 'supergrid' that refines each grid cell:
+        9---4---8
+        |       |
+        1   5   3
+        |       |
+        6---2---7
+        """
+        if self._sin_sg3 is None:
+            self._init_cell_trigonometry()
+        return self._sin_sg3
+
+    @property
+    def sin_sg4(self):
+        """
+        Sine of the angle at point 4 of the 'supergrid' that refines each grid cell:
+        9---4---8
+        |       |
+        1   5   3
+        |       |
+        6---2---7
+        """
+        if self._sin_sg4 is None:
+            self._init_cell_trigonometry()
+        return self._sin_sg4
+
+    @property
+    def sin_sg5(self):
+        """
+        Sine of the angle at point 5 of the 'supergrid' that refines each grid cell:
+        9---4---8
+        |       |
+        1   5   3
+        |       |
+        6---2---7
+        For the center point this is one minus the inner product of ec1 and ec2 squared
+        """
+        if self._sin_sg5 is None:
+            self._init_cell_trigonometry()
+        return self._sin_sg5
+
+    @property
+    def sin_sg6(self):
+        """
+        Sine of the angle at point 6 of the 'supergrid' that refines each grid cell:
+        9---4---8
+        |       |
+        1   5   3
+        |       |
+        6---2---7
+        """
+        if self._sin_sg6 is None:
+            self._init_cell_trigonometry()
+        return self._sin_sg6
+
+    @property
+    def sin_sg7(self):
+        """
+        Sine of the angle at point 7 of the 'supergrid' that refines each grid cell:
+        9---4---8
+        |       |
+        1   5   3
+        |       |
+        6---2---7
+        """
+        if self._sin_sg7 is None:
+            self._init_cell_trigonometry()
+        return self._sin_sg7
+
+    @property
+    def sin_sg8(self):
+        """
+        Sine of the angle at point 8 of the 'supergrid' that refines each grid cell:
+        9---4---8
+        |       |
+        1   5   3
+        |       |
+        6---2---7
+        """
+        if self._sin_sg8 is None:
+            self._init_cell_trigonometry()
+        return self._sin_sg8
+
+    @property
+    def sin_sg9(self):
+        """
+        Sine of the angle at point 9 of the 'supergrid' that refines each grid cell:
+        9---4---8
+        |       |
+        1   5   3
+        |       |
+        6---2---7
+        """
+        if self._sin_sg9 is None:
+            self._init_cell_trigonometry()
+        return self._sin_sg9
+
+    @property
+    def cosa(self):
+        """
+        The inner product of ee1 and ee2
```

Comment:
Cosine of the angle between coordinate lines (ie. \alpha in Putman and Lin 2007 or Harris and Lin 2013) at the cell corners, averaged to ensure consistent answers.